extensions/markdown-language-features/src/commands/openImage.ts

```diff
@@ -0,0 +1,22 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import { Command } from '../commandManager';
+import { MarkdownPreviewManager } from '../preview/previewManager';
+
+export class OpenImageCommand implements Command {
+	public readonly id = '_markdown.openImage';
+
+	public constructor(
+		private readonly _webviewManager: MarkdownPreviewManager,
+	) { }
+
+	public execute(args: { resource: string; imageSource: string }) {
+		const source = vscode.Uri.parse(args.resource);
+		const imageSourceUri = vscode.Uri.file(vscode.Uri.parse(args.imageSource).path);
```

Comment:
You should only need a single call to `vscode.Uri.parse`. Using  `vscode.Uri.file` again means it will only work if the workspace is on the local file system

If parse isn't enough, try seeing if you can pass the fill string uri to it. It should be something like: `file:///path/to/workspace/image.png`